func.go

```diff
@@ -81,6 +81,12 @@ func RegisterLibFunc(fptr interface{}, handle uintptr, name string) {
 // using unsafe.Slice. Doing this means that it becomes the responsibility of the caller to care about the lifetime
 // of the pointer
 //
+// # Structs
+//
+// Purego can handle the most common structs that have fields of builtin types like int8, uint16, float32, etc. However,
+// it does not _currently_ support aligning fields properly. It is therefore the responsibility of the caller to ensure
```

Comment:
The situation seems the same as Windows: adjusting the layout is a caller's responsibility. Do you think we should fix this?

Reply:
Perhaps, however it is not possible for purego to ensure it's doing the right thing since it can't see any C headers to know what the signature is.